community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/IndexSpecifier.java

```diff
@@ -58,21 +58,31 @@ public String toString()
         return specification;
     }
 
-    private Pair<String, String> parse()
+    private Pair<String,String[]> parse()
     {
-        Pattern pattern = Pattern.compile( "" +
-                ":" + or( simpleIdentifier( "simpleLabel" ), complexIdentifier( "complexLabel" ) ) +
-                "\\(" + or( simpleIdentifier( "simpleProperty" ), complexIdentifier( "complexProperty" ) ) + "\\)"
+        Pattern pattern = Pattern.compile(
+                ":" + or( identifier(), qoutedIdentifier() ) + // Match the label
+                "\\(" + or( identifier(), qoutedIdentifier() ) + // Match the first property
+                "(?:\\,\\s" + or( identifier(), qoutedIdentifier() ) + ")*\\)" // Match following properties
```

Comment:
You probably a want a `*` after the `\\s` for a more flexible syntax. Looking at the legacy open Cypher specification, space is also allowed between the `:` and the label identifier when creating indexes, though not when doing index lookups via the `START` clause.